hypothesis-python/src/hypothesis/internal/conjecture/data.py

```diff
@@ -645,16 +640,16 @@ def start_example(self, i: int, label_index: int) -> None:
 
     class _mutator_groups(ExampleProperty):
         def begin(self) -> None:
-            self.groups: "Dict[Tuple[int, int], List[int]]" = defaultdict(list)
+            self.groups: "Dict[Tuple[int, int], List[int]]" = defaultdict(set)
 
         def start_example(self, i: int, label_index: int) -> None:
-            depth = len(self.example_stack)
-            self.groups[label_index, depth].append(i)
+            key = (self.examples[i].ir_start, self.examples[i].ir_end)
+            self.groups[label_index].add(key)
 
```

Comment:
I'm not tracking uniqueness by depth anymore, so this should be strictly more powerful (= allows replacements in more places). Is there a reason depth was tracked before?

Reply:
I'm guessing performance - it might be cheaper to only consider same-depth swaps sometimes?  But if it's working well, I'm happy to stop doing that.

Reply:
let's keep doing this more powerful approach then.